### PR TITLE
[search] Added buttons to mark all results as Relevant or Irrelevant.

### PR DESCRIPTION
**Pull request by mpimenov**

### CODE REVIEW

File: search/search_quality/assessment_tool/edits.cpp [hunk 1]
```diff
@@ -2,6 +2,17 @@
 
 #include "base/assert.hpp"
 
+namespace
+{
+void UpdateNumEdits(Edits::Entry const & entry, Edits::MaybeRelevance const & r, size_t & numEdits)
+{
+  if (entry.m_curr != entry.m_orig && r == entry.m_orig)
+    --numEdits;
```
Comment (tatiana-kondakova) [inline search/search_quality/assessment_tool/edits.cpp]: опционально: можно предварительно чекать numEdits

Comment (mpimenov) [reply search/search_quality/assessment_tool/edits.cpp]: done

File: search/search_quality/assessment_tool/result_view.cpp [hunk 2]
```diff
@@ -167,3 +161,24 @@ void ResultView::OnRelevanceChanged()
...
+{
+  if (!m_editor)
+    return;
+
+  m_irrelevant->setChecked(false);
+  m_relevant->setChecked(false);
+  m_vital->setChecked(false);
+
+  auto const & r = m_editor->Get();
+  if (!r.m_unknown)
```
Comment (ygorshenin) [inline search/search_quality/assessment_tool/result_view.cpp]: How about to replace this MaybeRelevance by boost::optional<Relevance>? IMHO the code will be cleaner.

File: search/search_quality/assessment_tool/main_model.cpp [hunk 3]
```diff
@@ -383,6 +393,21 @@ void MainModel::ShowMarks(Context const & context)
...
+{
+  CHECK_GREATER_OR_EQUAL(m_selectedSample, 0, ());
+  CHECK_LESS(m_selectedSample, m_contexts.Size(), ());
+  auto & context = m_contexts[m_selectedSample];
+
+  context.m_foundResultsEdits.SetAllRelevances(relevance);
+  context.m_nonFoundResultsEdits.SetAllRelevances(relevance);
+
+  m_view->OnResultChanged(m_selectedSample, View::ResultType::Found, Edits::Update::MakeAll());
+  m_view->OnResultChanged(m_selectedSample, View::ResultType::NonFound, Edits::Update::MakeAll());
```
Comment (ygorshenin) [inline search/search_quality/assessment_tool/main_model.cpp]: What the reason to call OnResultChanged for both Found and NonFound?

File: search/search_quality/assessment_tool/result_view.cpp (hunk 2 above)
Comment (mpimenov) [reply search/search_quality/assessment_tool/result_view.cpp]: Yes, it's in the plans.

File: search/search_quality/assessment_tool/main_model.cpp (hunk 3 above)
Comment (mpimenov) [reply search/search_quality/assessment_tool/main_model.cpp]: My reasoning was "it won't hurt and will be harder to forget when refactoring".
Currently,
1. A click on the button modifies relevances both for found and non-found results, so let's inform both.
2. These two lines are essentially nop, the next two are enough.

File: search/search_quality/assessment_tool/result_view.cpp (hunk 2 above)
Comment (mpimenov) [reply search/search_quality/assessment_tool/result_view.cpp]: #8325

**Pull request merged**